tools/autograd/gen_python_functions.py

```diff
@@ -483,130 +453,166 @@ def is_noarg_binding(overloads):
 
 """)
 
-TORCH_FUNCTION_CHECK = CodeTemplate("""\
-if(_r.has_torch_function()) {
-  return handle_torch_function(_r, ${self_}, args, kwargs, ${namespace}, ${modulename});
-}
-""")
-
-TORCH_FUNCTION_CHECK_NOARGS = CodeTemplate("""\
-if(check_has_torch_function(self_)) {
-  return handle_torch_function(self_, ${name});
-}
-""")
-
-# NOTE: we type the unpacked self as Tensor not Variable to avoid return type
-# discrepancies on method resolution (e.g. Variable::detach_ returns void
-# rather than Tensor &)
-UNPACK_SELF = "Tensor& self = reinterpret_cast<THPVariable*>(self_)->cdata;"
-
-
-def method_impl(name, declarations, is_python_method, module):
+def method_impl(
+    name: str, module: str,
+    overloads: Tuple[PythonSignatureNativeFunctionPair, ...],
+    *,
+    noarg: bool, method: bool
+) -> str:
     """
     Generate a python binding for all overloads of an op.
     """
     pycname = get_pycname(name)
+    namedtuple_inits, namedtuple_typenames = emit_namedtuple_typedefs(overloads)
 
     method_header = ['HANDLE_TH_ERRORS']
-    method_header += emit_namedtuple_typedefs(declarations)
-    method_header += [UNPACK_SELF] if is_python_method else []
+    method_header += namedtuple_inits
 
-    method_footer = ['END_HANDLE_TH_ERRORS']
+    # NOTE: we type the unpacked self as Tensor not Variable to avoid return type
+    # discrepancies on method resolution (e.g. Variable::detach_ returns void
+    # rather than Tensor &)
```

Comment:
BTW, this comment is defunct now, because Tensor and Variable are the same in C++ now

Reply:
ok, I deleted this comment - not worth calling out anymore?